marklogic-data-hub-central/ui/src/components/results-tabular-view/results-tabular-view.tsx

```diff
@@ -43,42 +89,159 @@ const ResultsTabularView = (props) => {
         return columns;
     }
 
-    const tableHeaders = tableHeaderRender(selectedTableColumns);
+
+    const updatedTableHeader = () => {
+        let header = tableHeaderRender(selectedTableColumns);
+        let detailView = {
+            title: 'Detail View',
+            dataIndex: 'detailView',
+            key: '0-d'
+        }
+        header.push(detailView);
+        return header;
+    }
+
+    const tableHeaders = props.selectedEntities?.length === 0 ? DEFAULT_ALL_ENTITIES_HEADER : updatedTableHeader();
+
 
     const tableDataRender = (item) => {
         let dataObj = {};
+        let primaryKeyValue = item.primaryKey.propertyValue;
+        let isUri = item.primaryKey.propertyPath === 'uri';
+        let uri = encodeURIComponent(item.uri);
+        let path = { pathname: `/detail/${isUri ? '-' : primaryKeyValue}/${uri}` };
```

Comment:
I'm not clear on why the primaryKeyValue - which could be any string with any characters in it - would be used for anything except for displaying it. Why does this path differ based on whether the primary key is a URI or not?

Reply:
@rjrudin - This path property is just created to be sent over to the detail view page via the react Link router where its being used in a few places. Since this story was only about the navigation and not updating the detail view page so i didn't touched it.

Reply:
This seems like it's asking for trouble though - the primaryKeyValue could have a "/" in it, would the path still work?? 

I feel that using the primary key value for anything other than display purposes is risky. We can't guarantee that it's unique, and it could contain any character in it. Only the URI is guaranteed to be unique.

Reply:
@rjrudin - I don't think we are handling the case currently, if there is a "/" in the primary key. I am going to create a separate bug to handle that case and also to remove primary key logic completely from Detail , Detail header and other components as well (if any). We might need a bit of refactoring over there.

Reply:
A separate ticket works, thanks!